pyroma/ratings.py

```diff
@@ -266,13 +268,39 @@ def test(self, data):
 
     def message(self):
         if self._major_version_specified:
-            return (
-                "The classifiers should specify what minor versions of "
-                "Python you support as well as what major version."
-            )
+            return "The classifiers should specify what minor versions of "\
+                   "Python you support as well as what major version."
         return "You should specify what Python versions you support."
 
 
+class PythonRequiresVersion(BaseTest):
+    def test(self, data):
+        self.weight = 100
+        python_requires = data.get("python_requires", None)
+        requires_python = data.get("requires_python", None)
+
+        # python_requires and requires_python are turned into 'Requires-Python' header
+        # https://packaging.python.org/en/latest/guides/dropping-older-python-versions/#specify-the-version-ranges-for-supported-python-distributions
+        # Some tools seems to report only one (eg: test_complete, test_markdown fail when 'requires_python' is not checked).
+        # So using whichever has a value provided
+        python_requires = python_requires or requires_python
+
+        if not python_requires:
+            return False
+
+        try:
+            from packaging.specifiers import SpecifierSet
+            SpecifierSet(python_requires)
+            return True
+        except packaging.specifiers.InvalidSpecifier as e:
+            return False
```

Comment:
As a general best practice, your `try` block should be as minimal as possible and only the line(s) of code in which the error is actually expected to occur should be included within, otherwise you might swallow errors you don't intend. Also, its not clear why you're importing the necessary submodule down here inside the `try` block, instead of at the top like everything else (whereas `packaging` is already imported at the top level, but never used).

Therefore, you should do (as in my example)

```suggestion
        try:
            packaging.specifiers.SpecifierSet(python_requires)
        except packaging.specifiers.InvalidSpecifier as e:
            return False
        return True
```